Settings redesign

Bilb opened the pull request.

Bilb committed:
fix: remove logs of HF  19.0 and 19.1 values

Bilb committed:
feat: start of settings screen redesign

Bilb committed:
fix: update copy of settings

Bilb committed:
fix: batch rename margins to their var equivalent

Some of those are used in calculus and cannot be renamed

Bilb committed:
chore: move SessionToggle & SettingList to styled

Bilb committed:
chore: move onion dialog components to styled

Bilb committed:
fix: add new designs of typing message settings with animation

Bilb committed:
chore: keep full stop for everything in settings description

Bilb committed:
feat: update design of the Notification Settings screen

the preview button is not linked yet

Bilb committed:
feat: add the preview button of notifications in settings

Bilb committed:
Merge branch 'menu-redesign' into settings-redesign

yougotwill commented on ts/components/basic/SessionToggle.tsx:
```diff
@@ -45,14 +45,38 @@ export const SessionToggle = (props: Props) => {
   );
 };
+
+const StyledKnob = styled.div<{ active: boolean }>`
```
Styled components should be above

yougotwill commented on ts/components/dialog/OnionStatusPathDialog.tsx:
```diff
@@ -84,26 +98,45 @@ const OnionPathModalInner = () => {
+          </StyledLightsContainer>
           <Flex container={true} flexDirection="column" alignItems="flex-start">
-            {nodes.map((snode: Snode | any, index: number) => {
+            {nodes.map((snode: Snode | any) => {
```
if we are using map shouldn't the react props have a key property? https://reactjs.org/docs/lists-and-keys.html

Bilb replied:
the key was actually inserted in the element returned a few lines below in `OnionCountryDisplay`, but yeah I moved the setting of the key here cause it makes more sense.

yougotwill commented on ts/components/settings/SessionSettingListItem.tsx:
```diff
@@ -1,59 +1,120 @@
+        description={description}
+        childrenDescription={childrenDescription}
+      />
+      <div>{children}</div>
```
Don't think we need the div tags here? For Example

```
export const SessionTheme = ({ children }: { children: any }) => (
  <>
    <SessionGlobalStyles />
    {children}
  </>
);
```

yougotwill commented on ts/components/settings/SessionSettings.tsx:
```diff
@@ -42,22 +48,62 @@ interface State {
-    </div>
+      </StyledSpanSessionInfo>
+      <StyledSpanSessionInfo>{window.versionInfo.commitHash}</StyledSpanSessionInfo>
+    </StyledVersionInfo>
```
Nice!

yougotwill commented on ts/components/settings/SessionSettings.tsx:
```diff
@@ -89,6 +136,42 @@ export const PasswordLock = ({
+      return <SessionNotificationGroupSettings hasPassword={hasPassword} />;
+    case SessionSettingCategory.Privacy:
+      return (
+        <SettingsCategoryPrivacy onPasswordUpdated={onPasswordUpdated} hasPassword={hasPassword} />
```
Did prettier auto insert these brackets?

yougotwill commented on ts/components/settings/SessionSettingsHeader.tsx:
```diff
@@ -1,15 +1,60 @@
   );
 };
+
+const StyledSettingsHeader = styled.div`
```
Move these above the main component

yougotwill commented on ts/components/settings/section/CategoryAppearance.tsx:
```diff
@@ -1,72 +1,25 @@
-
     return (
       <>
+        {/* add theme switching here */}
```
Could you add TODO infront please 🙏

yougotwill commented on ts/components/settings/section/CategoryHelp.tsx:
```diff
@@ -0,0 +1,37 @@
+import { SessionSettingButtonItem } from '../SessionSettingListItem';
+
+export const SettingsCategoryHelp = (props: { hasPassword: boolean | null }) => {
+  if (props.hasPassword !== null) {
```
This could be? But as you want

```
  return props.hasPassword !== null ? (
    <>
      <SessionSettingButtonItem
        title={window.i18n('surveyTitle')}
        onClick={() => void shell.openExternal('https://getsession.org/survey')}
        buttonColor={SessionButtonColor.Primary}
        buttonType={SessionButtonType.Square}
        buttonText={window.i18n('goToOurSurvey')}
      />
      <SessionSettingButtonItem
        title={window.i18n('helpUsTranslateSession')}
        onClick={() => void shell.openExternal('https://crowdin.com/project/session-desktop/')}
        buttonColor={SessionButtonColor.Primary}
        buttonType={SessionButtonType.Square}
        buttonText={window.i18n('translation')}
      />
      <SessionSettingButtonItem
        onClick={() => {
          ipcRenderer.send('show-debug-log');
        }}
        buttonColor={SessionButtonColor.Primary}
        buttonType={SessionButtonType.Square}
        buttonText={window.i18n('showDebugLog')}
      />
    </>
  ) : null;
```

Bilb replied:
I like early returns when I can (when no hooks are called below). I find it easier to read :)

yougotwill commented on ts/components/settings/section/CategoryPermissions.tsx:
```diff
@@ -0,0 +1,98 @@
+  const forceUpdate = useUpdate();
+  const isStartInTrayActive = Boolean(window.getSettingValue(SettingsKey.settingsStartInTray));
+
+  if (props.hasPassword !== null) {
```
Could be one return as I showed above

yougotwill commented on ts/components/settings/section/CategoryPrivacy.tsx:
```diff
@@ -50,105 +26,82 @@ function displayPasswordModal(
+  return (
+    <>
+      <SpacerLG />
+
```
Delete new line

Bilb committed:
refactor: fix PR reviews

Bilb committed:
Merge remote-tracking branch 'upstream/clearnet' into settings-redesign

Bilb committed:
feat: update settings Help page with new items and links

Bilb committed:
chore: move typing animation to styled components

Bilb committed:
fix: make use of useSet to select in memberList

Bilb committed:
feat: add the BlockedList component in Settings

Bilb committed:
feat: add theme switcher +accent color from settings

not all of the logic is linked and not all of the colors are there as this is part of theming

Bilb committed:
fix: radio button in settings + registration page

Bilb committed:
Merge remote-tracking branch 'upstream/clearnet' into settings-redesign

Bilb committed:
Merge branch 'clearnet' into settings-redesign
